---
title: "CUT&RUN LPM WT vs KO - Figures"
author: "Domien Vanneste"
date: "`r format (Sys.time(), format='%Y-%m-%d %H:%M:%S %z')`"
output:
  pdf_document:
    toc: yes
    toc_depth: 2
  header-includes: "-| ```{=latex} \\usepackage{fvextra} \\DefineVerbatimEnvironment{Highlighting}{Verbatim}{
    breaksymbolleft={}, showspaces = false, showtabs = false, breaklines, commandchars=\\\\\\{\\}
    } ```"
  html_document:
    toc: yes
    toc_depth: '2'
    df_print: paged
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(tidy.opts = list(width.cutoff = 60), tidy = TRUE)
```

\newpage

# Introduction
To test whether MafB also regulates mac identity in primary RTM, we performed CUT&RUN for MafB on LPMs from Mafbfl/fl or Lyz2CreMafbfl/fl mice. CUT&RUN was performed with a CUTANA ChIC/CUT&RUN Kit (EpiCypher, 141048) according to manufacturer’s instructions, with modifications. BMDMs from Mafbfl/fl or Lyz2CreMafbfl/fl mice were washed with ice-cold PBS and fixed with 0.1 % formaldehyde in PBS for 2 min at room temperature. Fixation was quenched by adding glycine (Merck, 104691000) to 0.125 M.  For each CUT&RUN sample, 1 x 10^6 fixed cells and 0.5 µg antibodies were added:

IgG (EpiCypher, 13-0042)
MafB2 (Cell Signaling Technology, 41019)

CUT&RUN libraries were prepared with a CUTANA™ CUT&RUN Library Prep Kit (EpiCypher, 141001) according to manufacturer’s instructions. These libraries were sequenced on an NovaSeq 6000 (Illumina) sequencer on an S4 flow cell at 10 million reads per sample.
\newpage

# Load Packages

```{r}
suppressMessages({
library(ggplot2)
library(colorRamp2)
library(ComplexHeatmap)
library(readxl)
library(futile.logger)
library(grid)
library(rtracklayer)
library(dplyr)
library(VennDiagram)
library(Rtoolbox)
})
```


# Figure S6A

```{r}
MafB2_peaks_rep1 <- read.table("MafB2_WT_R1.seacr.peaks.stringent.bed", header = FALSE, sep = "\t")

MafB2_peaks_rep2 <- read.table("MafB2_WT_R2.seacr.peaks.stringent.bed", header = FALSE, sep = "\t")

MafB2_peaks_consensus <- read.table("MafB2_WT.seacr.consensus.peak_counts.bed", header = FALSE, sep = "\t")
```

```{r}
#peaks rep1
nrow(MafB2_peaks_rep1)

#peaks rep2
nrow(MafB2_peaks_rep2)

#consensus peaks
nrow(MafB2_peaks_consensus)

#shared peaks
length(which(MafB2_peaks_consensus[,10] == 2))
```

## Rep. 1

```{bash eval = FALSE}
annotatePeaks.pl MafB2_WT_R1.seacr.peaks.stringent.bed mm10 -size 4000 -hist 10 -d MafB2_WT_R1/ > MafB2_WT_R1_hist.txt
```

```{r}
MafB2_WT_R1_hist <- read.table("MafB2_WT_R1_hist.txt", header = TRUE, sep = "\t")

colnames(MafB2_WT_R1_hist)[colnames(MafB2_WT_R1_hist) == "Distance.from.Center..cmd.annotatePeaks.pl.MafB2_WT_R1.seacr.peaks.stringent.bed.mm10..size.4000..hist.10..d.MafB2_WT_R1.."] <- "Distance.from.Center"
```

n peaks : 24776
FDR =  0.00230583412634511

```{r fig.height=3, fig.width=3}
ggplot(data = MafB2_WT_R1_hist, aes(x = Distance.from.Center, y = MafB2_WT_R1..Coverage))+
  geom_line(show.legend = FALSE, colour = "blue")+
  ggtitle("Rep. 1: 24776 peaks \nFDR = 0.002305834")+
  xlab("Distance from peak center (bp)")+
  ylab("Normalized tags")+
  theme_classic()+
  theme(axis.text.x = element_text(color = "black"), axis.text.y = element_text(color = "black"), axis.ticks.length=unit(.15, "cm"), panel.border = element_rect(fill=NA,color="black", linetype="solid"))
```

```{r}
ggsave("Figure_S6E_Rep1.pdf", width = 3, height = 3)
```


## Rep. 2

```{bash eval = FALSE}
annotatePeaks.pl MafB2_WT_R2.seacr.peaks.stringent.bed mm10 -size 4000 -hist 10 -d MafB2_WT_R2/ > MafB2_WT_R2_hist.txt
```

```{r}
MafB2_WT_R2_hist <- read.table("MafB2_WT_R2_hist.txt", header = TRUE, sep = "\t")

colnames(MafB2_WT_R2_hist)[colnames(MafB2_WT_R2_hist) == "Distance.from.Center..cmd.annotatePeaks.pl.MafB2_WT_R2.seacr.peaks.stringent.bed.mm10..size.4000..hist.10..d.MafB2_WT_R2.."] <- "Distance.from.Center"
```

n peaks : 15978
FDR =  0.0057846571008926

```{r fig.height=3, fig.width=3}
ggplot(data = MafB2_WT_R2_hist, aes(x = Distance.from.Center, y = MafB2_WT_R2..Coverage))+
  geom_line(show.legend = FALSE, colour = "blue")+
  ggtitle("Rep. 2: 15,978 peaks \nFDR = 0.005784657")+
  xlab("Distance from peak center (bp)")+
  ylab("Normalized tags")+
  theme_classic()+
  theme(axis.text.x = element_text(color = "black"), axis.text.y = element_text(color = "black"), axis.ticks.length=unit(.15, "cm"), panel.border = element_rect(fill=NA,color="black", linetype="solid"))
```

```{r}
ggsave("Figure_S6E_Rep2.pdf.pdf", width = 3, height = 3)
```

# Figure S6F

```{bash eval = FALSE}
samtools view -c -F 4 MafB2_WT_R1.target.markdup.sorted.bam
bedtools intersect -a MafB2_WT_R1.target.markdup.sorted.bam -b MafB2_WT_R1.seacr.peaks.stringent.bed -bed | wc -l

samtools view -c -F 4 MafB2_WT_R2.target.markdup.sorted.bam
bedtools intersect -a MafB2_WT_R2.target.markdup.sorted.bam -b MafB2_WT_R2.seacr.peaks.stringent.bed -bed | wc -l
```


# Figure S6G

```{r}
MafB2_peaks_merged <- MafB2_peaks_consensus[MafB2_peaks_consensus[,10] == 2, ]

split_scores <- strsplit(as.character(MafB2_peaks_merged $V6), ",")

means <- sapply(split_scores, function(x) {
  if (all(is.na(x))) return(NA_real_)
  nums <- as.numeric(x)
  mean(nums, na.rm = TRUE)
})

MafB2_peaks_merged$V4 <- means

write.table(MafB2_peaks_merged, "MafB2_peaks_merged.bed", sep="\t", quote = FALSE, row.names = FALSE, col.names = FALSE)
```

```{bash eval = FALSE}
annotatePeaks.pl MafB2_peaks_merged.bed mm10 -d MafB2_WT_R1/ MafB2_WT_R2/ MafB2_KO_R1/ MafB2_KO_R2/ > MafB2_counts.txt
```

```{r}
MafB2_counts <- read.table("MafB2_counts.txt", header = TRUE, sep = "\t")

colnames(MafB2_counts)[colnames(MafB2_counts) == "PeakID..cmd.annotatePeaks.pl.MafB2_peaks_merged.bed.mm10..d.MafB2_WT_R1..MafB2_WT_R2..MafB2_KO_R1..MafB2_KO_R2.."] <- "PeakID"

colnames(MafB2_counts)[colnames(MafB2_counts) == "MafB2_WT_R1..Tag.Count.in.given.bp..5198073.0.Total..normalization.factor...1.92..effective.total...10000000."] <- "MafB2_WT_R1"

colnames(MafB2_counts)[colnames(MafB2_counts) == "MafB2_WT_R2..Tag.Count.in.given.bp..5587453.0.Total..normalization.factor...1.79..effective.total...10000000."] <- "MafB2_WT_R2"

colnames(MafB2_counts)[colnames(MafB2_counts) == "MafB2_KO_R1..Tag.Count.in.given.bp..6219029.0.Total..normalization.factor...1.61..effective.total...10000000."] <- "MafB2_KO_R1"

colnames(MafB2_counts)[colnames(MafB2_counts) == "MafB2_KO_R2..Tag.Count.in.given.bp..6260374.0.Total..normalization.factor...1.60..effective.total...10000000."] <- "MafB2_KO_R2"
```

```{r fig.height=3.225, fig.width=3}
ggplot(data = MafB2_counts, aes(x =log2(MafB2_WT_R1+1), y =log2(MafB2_WT_R2+1)))+
  geom_point(show.legend = FALSE, colour = "blue")+
  geom_abline(slope = 1)+
  ggtitle("11000 merged peaks")+
  xlab("Log2(norm tags + 1) \nRep. 1")+
  ylab("Rep. 2 \nLog2(norm tags + 1)")+
  xlim(0,11)+
  ylim(0,11)+
  theme_classic()+
  theme(axis.text.x = element_text(color = "black"), axis.text.y = element_text(color = "black"), axis.ticks.length=unit(.15, "cm"), panel.border = element_rect(fill=NA,color="black", linetype="solid"))
```

```{r}
ggsave("Figure_S6G.pdf", width = 3, height = 3.225)
```

# Figure 6I

## Histogram

```{bash eval = FALSE}
annotatePeaks.pl MafB2_peaks_merged.bed mm10 -size 4000 -hist 10 -d MafB2_WT_R1/ MafB2_WT_R2/ MafB2_KO_R1/ MafB2_KO_R2/ > MafB2_hist.txt
```

```{r}
MafB2_hist <- read.table("MafB2_hist.txt", header = TRUE, sep = "\t")

colnames(MafB2_hist)[colnames(MafB2_hist) == "Distance.from.Center..cmd.annotatePeaks.pl.MafB2_peaks_merged.bed.mm10..size.4000..hist.10..d.MafB2_WT_R1..MafB2_WT_R2..MafB2_KO_R1..MafB2_KO_R2.."] <- "Distance.from.Center"

MafB2_hist$MafB2_WT <- rowMeans(MafB2_hist[,c('MafB2_WT_R1..Coverage', 'MafB2_WT_R2..Coverage')])

MafB2_hist$MafB2_KO <- rowMeans(MafB2_hist[,c('MafB2_KO_R1..Coverage', 'MafB2_KO_R2..Coverage')])
```

```{r fig.height=4, fig.width=3}
ggplot(data = MafB2_hist, aes(x = Distance.from.Center, y = MafB2_WT))+
  geom_line(show.legend = FALSE, colour = "blue")+
  geom_line(show.legend = FALSE, colour = "red", aes(x = Distance.from.Center, y = MafB2_KO))+
  ggtitle("MafB CUT&RUN in LPM")+
  xlab("Distance from peak center (bp)")+
  ylab("Normalized tags")+
  theme_classic()+
  theme(axis.text.x = element_text(color = "black"), axis.text.y = element_text(color = "black"), axis.ticks.length=unit(.15, "cm"), panel.border = element_rect(fill=NA,color="black", linetype="solid"))
```

```{r}
ggsave("Figure_6I_WTvsKO_Hist.pdf", width = 3, height = 4)
```

## Heatmap

```{bash eval = FALSE}
annotatePeaks.pl MafB2_peaks_merged.bed mm10 -size 4000 -hist 25 -ghist -d MafB2_WT_R1/ MafB2_WT_R2/ MafB2_KO_R1/ MafB2_KO_R2/ > MafB2_heatmap.txt
```

```{r}
MafB2_heatmap <- read.table("MafB2_heatmap.txt", header = TRUE, check.names = FALSE, sep = "\t")

MafB2_WT_R1_heatmap <- MafB2_heatmap[, 2:162]
MafB2_WT_R2_heatmap <- MafB2_heatmap[, 163:323]
MafB2_WT_heatmap <- (MafB2_WT_R1_heatmap+MafB2_WT_R2_heatmap)/2

MafB2_KO_R1_heatmap <- MafB2_heatmap[, 324:484]
MafB2_KO_R2_heatmap <- MafB2_heatmap[, 485:645]
MafB2_KO_heatmap <- (MafB2_KO_R1_heatmap+MafB2_KO_R2_heatmap)/2
```

### WT

```{r fig.height=4, fig.width=2}
col_fun = colorRamp2(c(0, 10), c( "white", "red"))

pdf("Figure_6I_heatmap_WT.pdf", width = 2, height = 4)
Heatmap(MafB2_WT_heatmap, col_fun, cluster_columns = F, cluster_rows = F, heatmap_legend_param = list(title = "Tags"), use_raster = TRUE, raster_quality = 10, column_title = "Mafbfl/fl")
dev.off()
```

### KO

```{r fig.height=4, fig.width=2}
col_fun = colorRamp2(c(0, 10), c( "white", "red"))

pdf("Figure_6I_heatmap_KO.pdf", width = 2, height = 4)
Heatmap(MafB2_KO_heatmap, col_fun, cluster_columns = F, cluster_rows = F, heatmap_legend_param = list(title = "Tags"), use_raster = TRUE, raster_quality = 10, , column_title = "Lyz2CreMafbfl/fl")
dev.off()
```

# Figure 6J

```{r}
MafB2_counts$MafB2_WT <- rowMeans(MafB2_counts[,c('MafB2_WT_R1', 'MafB2_WT_R2')])
MafB2_counts$MafB2_KO <- rowMeans(MafB2_counts[,c('MafB2_KO_R1', 'MafB2_KO_R2')])

MafB2_counts$log2_MafB2_WT <- log2(MafB2_counts$MafB2_WT + 1)
MafB2_counts$log2_MafB2_KO <- log2(MafB2_counts$MafB2_KO + 1)
MafB2_counts$below_diag <- MafB2_counts$log2_MafB2_KO < MafB2_counts$log2_MafB2_WT
```

```{r fig.height=3.225, fig.width=3}
ggplot(data = MafB2_counts, aes(x = log2_MafB2_WT, y = log2_MafB2_KO)) +
  geom_point(aes(color = below_diag), show.legend = FALSE) +
  scale_color_manual(values = c("FALSE" = "red", "TRUE" = "blue")) +
  geom_abline(slope = 1) +
  ggtitle("11000 peaks") +
  xlab("Log2(norm tags + 1) \nMafbfl/fl") +
  ylab("Lyz2CreMafbfl/fl \nLog2(norm tags + 1)") +
  xlim(0, 11) +
  ylim(0, 11) +
  theme_classic() +
  theme(
    axis.text.x = element_text(color = "black"),
    axis.text.y = element_text(color = "black"),
    axis.ticks.length = unit(.15, "cm"),
    panel.border = element_rect(fill = NA, color = "black", linetype = "solid")
  )
```

```{r}
ggsave("Figure_6J.pdf", width = 3, height = 3.225)
```

# Figure 6K

```{bash eval = FALSE}
annotatePeaks.pl MafB2_peaks.bed genome.fa -gtf genes.gtf > MafB2_peaks_annot.txt
```

```{r}
MafB2_peaks_annot <- read.table("MafB2_peaks_annot.txt", header = TRUE, sep = "\t")
```

```{r}
MafB2_peaks_annot$Annotation.not.detailed <- sub(" .*", "", MafB2_peaks_annot$Annotation)

table(MafB2_peaks_annot$Annotation.not.detailed)
```


# Figure 6L

```{r}
MafB2_BMDM_peaks <- read.table("MafB2_BMDM_peaks.bed", header = FALSE, sep = "\t")
MafB2_peaks <- read.table("MafB2_peaks.bed", header = FALSE, sep = "\t")

nrow(MafB2_BMDM_peaks)

nrow(MafB2_peaks)
```

```{bash eval=FALSE}
bedtools window -a MafB2_peaks.bed -b MafB2_BMDM_peaks.bed -w 100 > MafB2_overlap.bed
```

```{r}
MafB2_overlap <- read.table("MafB2_overlap.bed", header = FALSE, sep = "\t")

nrow(MafB2_overlap)
```

```{r fig.height=4, fig.width=4}
pdf(file = "Figure_6L_Venn.pdf", width = 4, height = 4)
draw.pairwise.venn(area1 = 7242, area2 = 10858, cross.area = 5133, category = c("BMDM", "LPM"), fill = c("red", "green"))
grid.newpage()
```

# Figure 6M

```{r}
# Helper function to display Venn diagram
display_venn <- function(x, ...){
  grid.newpage()
  venn_object <- venn.diagram(x, filename = NULL, ...)
  grid.draw(venn_object)
}
```

```{r}
MafB2_LPM_genes <- MafB2_peaks_annot[MafB2_peaks_annot[, 'Distance.to.TSS'] >= -2000 & MafB2_peaks_annot[, 'Distance.to.TSS'] <= 2000, ]

MafB2_LPM_genes <- unique(MafB2_LPM_genes$Gene.Name)
MafB2_LPM_genes <- na.omit(MafB2_LPM_genes)
```

```{r}
MafB2_BMDM_peaks_annot <- read.table("MafB2_BMDM_peaks_annot.txt", header = TRUE, sep = "\t")

MafB2_BMDM_genes <- MafB2_BMDM_peaks_annot[MafB2_BMDM_peaks_annot[, 'Distance.to.TSS'] >= -2000 & MafB2_BMDM_peaks_annot[, 'Distance.to.TSS'] <= 2000, ]

MafB2_BMDM_genes <- unique(MafB2_BMDM_genes$Gene.Name)
MafB2_BMDM_genes <- na.omit(MafB2_BMDM_genes)
```

```{r fig.height=4, fig.width=4}
x <- list(LPM = MafB2_LPM_genes, BMDM = MafB2_BMDM_genes)
pdf(file = "Figure_6M_Venn.pdf", width = 4, height = 4)
display_venn(x, fill = c("green","red"))
dev.off()
```

