---
title: "RNA-Seq BMDM WT vs KO"
author: "Domien Vanneste"
date: "`r format (Sys.time(), format='%Y-%m-%d %H:%M:%S %z')`"
output:
  pdf_document:
    toc: yes
    toc_depth: 2
  header-includes: "-| ```{=latex} \\usepackage{fvextra} \\DefineVerbatimEnvironment{Highlighting}{Verbatim}{
    breaksymbolleft={}, showspaces = false, showtabs = false, breaklines, commandchars=\\\\\\{\\}
    } ```"
  html_document:
    toc: yes
    toc_depth: '2'
    df_print: paged
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(tidy.opts = list(width.cutoff = 60), tidy = TRUE)
```

\newpage

# Introduction

Two hundred fifty thousand bone marrow monocytes from Mafbfl/fl or Lyz2CreMafbfl/fl mice were isolated, seeded in tissue culture-treated 6-well plates (Greiner, 657160) and differentiated into BMDM, as described above. After 4 days of differentiation, non-adherent cells were washed off with ice-cold PBS and total RNA was isolated with the standard TRIzol (Invitrogen, 15596018) RNA extraction protocol. Total RNA was further purified and concentrated using a RNA Clean & Concentrator Kit (Zymo Research, R1013) according to manufacturerâ€™s instructions. RNA quality and quantity were evaluated using a 5200 Fragment Analyzer (Agilent). Samples with a RIN > 9.9 were selected for sequencing. One hundred nanograms of RNA was used to generate the libraries using the TruSeq Stranded mRNA kit (Illumina, 20020594). These libraries were sequenced on an Illumina NovaSeq sequencer on an SP flow cell. Preprocessing, alignment to the mouse genome (GRCm38/mm10), sequence counting, and quality control of the bulk RNA-Seq data were carried out with the nf-core/rnaseq pipeline. RNA-Seq data were further analyzed using R Bioconductor (4.2.3) and the DESeq2 package (1.38.3).

```{r}
suppressMessages({
library(DESeq2)
library(ggplot2)
library(pheatmap)
library(RColorBrewer)
library(apeglm)
library(readxl)
library(ggrepel)
library(dplyr)
library(ComplexHeatmap)
})
```


# Counting from fastq data using nf-core/rnaseq pipeline

The following codes were used to do the mapping and counting. 
```{bash eval=FALSE}
nextflow run nf-core/rnaseq --input samplesheet.csv --fasta GRCm38/fasta/genome.fa --gtf GRCm38/genes/genes.gtf --outdir /Bulk_RNAseq/ -profile docker
```

`samplesheet.csv` is text file with 4 columns: sample, fastq_1, fastq_2 and strandedness. Prepared following to the software's instructions. 

# Counts data processing

```{r}
COUNTS <- read.table("salmon.merged.gene_counts.tsv",sep="\t", header=T, row.names = NULL)

dim(COUNTS)
```

```{r}
Genes <- COUNTS$gene_name

rownames(COUNTS) = make.names(Genes, unique=TRUE)

COUNTS <- COUNTS[,-c(1,2)]

COUNTS <- round(COUNTS, digits = 0)

head(COUNTS, 3)

COUNTS <- COUNTS[,c(5,6,7,8,1,2,3,4)]

head(COUNTS, 3)
```

```{r}
write.csv(COUNTS, file = "COUNTS.csv")
```

# Make metadata for bulkRNAseq samples 

```{r}
SampleSheet <- data.frame(
  `genotype` = c(rep("WT", 4), rep("KO", 4)),
  `gender` = c(rep(rep(c("female", "male"), each = 2), 2)),
  `replicate` = c(rep(rep(c("1", "2")), 4))
)

rownames(SampleSheet) <- colnames(COUNTS)

SampleSheet
```

# DESeq2 analysis

```{r}
dds <- DESeqDataSetFromMatrix(
  countData= COUNTS,
  colData= SampleSheet,
  design= ~ genotype + gender
)
dds
```

## Perform rlog transformation for distances and PCA

```{r}
# keep only genes with more than a single read
dds <- dds[ rowSums(counts(dds)) > 1,]
# perform rlog transformation for distances (for clustering) and PCA
rld<-rlog(dds)
```

```{r}
dds <- dds[ rowSums(counts(dds)) > 1,]
nrow(dds)
```


Calculate sample-to-sammple distances

```{r}
sampleDists <- dist( t( assay(rld) ) )
sampleDistMatrix <- as.matrix( sampleDists ) 
```


## Heatmap

```{r fig.width=5.5, fig.height=5}
colors <- colorRampPalette( rev(brewer.pal(ncol(COUNTS), "Blues")) )(255) 
heatmap <- pheatmap(sampleDistMatrix,
                    clustering_distance_rows=sampleDists,
                    clustering_distance_cols=sampleDists,
                    col=colors
)
heatmap
```

## PCA analysis

Calculate PCs: 
```{r}
PCA <- plotPCA(rld, intgroup = c("genotype", "gender"), returnData = TRUE)
```

```{r fig.height=4, fig.width=5}
percentVar<-round(100 * attr(PCA, "percentVar"))

ggplot(PCA,aes(PC1,PC2)) + 
	geom_point(size=3, aes(color = genotype, shape = gender)) +
  scale_color_manual(breaks = c("WT", "KO"), values = c("blue", "red")) +
  scale_shape_manual( breaks = c("female", "male"),values = c(16,1))+
	xlab(paste0("PC1: ", percentVar[1],"% variance")) +
	ylab(paste0("PC2: ", percentVar[2],"% variance"))+
  theme_classic()+
  theme(axis.text.x = element_text(color = "black"), axis.text.y = element_text(color = "black"), axis.ticks.length=unit(.15, "cm"), panel.border = element_rect(fill=NA,color="black", linetype="solid"))
```

## DEG analysis

```{r}
dds <- DESeq(dds)
resultsNames(dds)

dds$genotype <- relevel(dds$genotype, ref = "WT")
resultsNames(dds)
```

```{r}
dds <- DESeq(dds)
resultsNames(dds)

dds$genotype <- relevel(dds$genotype, ref = "KO")
resultsNames(dds)
```

```{r}
plotDispEsts(dds)
```

```{r}
results <- results(dds, name= "genotype_KO_vs_WT")
summary(results) 
```

```{r fig.height=4, fig.width=5}
plotMA(results, ylim = c(-6,9))
```

```{r}
results_shrunk <- lfcShrink(dds, coef = "genotype_KO_vs_WT", type = "apeglm")

results_shrunk$log2FoldChange <- results_shrunk$log2FoldChange*-1

deg <- merge(x=as.data.frame(results), y=as.data.frame(results_shrunk), by=c(0,1))
```

```{r fig.height=4, fig.width=5}
plotMA(results_shrunk, ylim = c(-6,9))
```

```{r}
Genes2 <- deg$Row.names

rownames(deg) = make.names(Genes2, unique=TRUE)

deg<- deg[,-1]

```


Filter
```{r}
deg <- deg[!is.na(deg$padj.y),]

deg_WT <- deg[deg$log2FoldChange.y< -1 & deg$padj.y<0.05,]
deg_KO <- deg[deg$log2FoldChange.y> 1 & deg$padj.y<0.05,]

deg_significant <- rbind(deg_WT, deg_KO)
```


```{r}
deg_ordered <- deg_significant[order(deg_significant$log2FoldChange.y) , ]
```

```{r}
write.csv(deg_ordered, file = "DEG.csv")
```

# Expression plot

```{r fig.height=4, fig.width=4}

COUNTS_LogTPM <- read_excel("COUNTS_LogTPM.xlsx")
COUNTS_LogTPM <- as.data.frame(COUNTS_LogTPM)


rownames(COUNTS_LogTPM) <- make.names(COUNTS_LogTPM$gene_name, unique=TRUE)

COUNTS_LogTPM$diffexpressed <- "Not significant"

KO <- rownames(deg_significant[deg_significant$log2FoldChange.y> 1,])
WT <- rownames(deg_significant[deg_significant$log2FoldChange.y< -1,])

COUNTS_LogTPM$diffexpressed[COUNTS_LogTPM$gene_name %in% WT] <- "WT"
COUNTS_LogTPM$diffexpressed[COUNTS_LogTPM$gene_name %in% KO] <- "KO"


COUNTS_LogTPM$label <- NA

Mo_signature_genes <- c("Ccr2", "Vcan", "Ly6c2", "Irf4","Itga1")

Mac_signature_genes <- c("Mafb", "C1qa", "C1qb", "C1qc", "Mertk", "Folr2")

COUNTS_LogTPM$label[COUNTS_LogTPM$gene_name %in% Mo_signature_genes] <- COUNTS_LogTPM$gene_name[COUNTS_LogTPM$gene_name %in% Mo_signature_genes]

COUNTS_LogTPM$label[COUNTS_LogTPM$gene_name %in% Mac_signature_genes] <- COUNTS_LogTPM$gene_name[COUNTS_LogTPM$gene_name %in% Mac_signature_genes]

ggplot(data = COUNTS_LogTPM, aes(x = WT_Log2_mean, y = KO_Log2_mean, col = diffexpressed, label = label))+
  geom_point(show.legend = FALSE)+
  geom_text_repel(min.segment.length = 0, box.padding = 0.5, size = 4, col = "black")+
  scale_color_manual(values = c("red", "grey", "blue"))+
  xlab("Log2(TPM + 1) \nMafbfl/fl")+
  ylab("Lyz2 Mafbfl/fl \nLog2(TPM + 1)")+
  theme_classic()+
  theme(axis.text.x = element_text(color = "black"), axis.text.y = element_text(color = "black"), axis.ticks.length=unit(.15, "cm"), panel.border = element_rect(fill=NA,color="black", linetype="solid"))

```


# Volcano plot

```{r fig.height=5, fig.width=4}

mat <- deg

mat$genes <- rownames(mat)

mat$diffexpressed <- "Not significant"

mat$diffexpressed[mat$log2FoldChange.y> 1 & mat$padj.y<0.05] <- "KO"

mat$diffexpressed[mat$log2FoldChange.y< -1 & mat$padj.y<0.05] <- "WT"



mat$label <- NA

Mo_signature_genes <- c("Ccr2", "Vcan", "Ly6c2", "Irf4","Itga1")

Mac_signature_genes <- c("Mafb", "C1qa", "C1qb", "C1qc", "Mertk", "Folr2")

mat$label[mat$genes %in% Mo_signature_genes] <- mat$genes[mat$genes %in% Mo_signature_genes]

mat$label[mat$genes %in% Mac_signature_genes] <- mat$genes[mat$genes %in% Mac_signature_genes]

ggplot(data = mat, aes(x = log2FoldChange.y, y = -log10(padj.y), col = diffexpressed, label = label))+
  geom_point(show.legend = FALSE)+
  geom_text_repel(min.segment.length = 0,box.padding = 0.5, size = 4, col = "black")+
  geom_vline(xintercept = c(-1,1), col = "black", linetype = "dotted")+
  geom_hline(yintercept = -log10(0.05), col = "black", linetype = "dotted")+
  scale_color_manual(values = c("red", "grey", "blue"))+
  xlab("Log2(Fold change)")+
  ylab("-Log10(Adjusted P value)")+
  xlim(-7, 7)+
  theme_classic()+
  theme(axis.text.x = element_text(color = "black", size = 12), axis.text.y = element_text(color = "black", size = 12), axis.ticks.length=unit(.25, "cm"), panel.border = element_rect(fill=NA,color="black", linetype="solid"))
```



# Heatmap

```{r fig.height=4, fig.width=3}
counts_norm <- counts(dds, normalized = TRUE)[rownames(deg_significant),]

counts_Z_score <- t(apply(counts_norm, 1, scale))

colnames(counts_Z_score) <- colnames(counts_norm)

counts_Z_score_ordered <- counts_Z_score[rownames(deg_ordered),] 

Heatmap(counts_Z_score_ordered, cluster_columns = FALSE, cluster_rows = FALSE, row_names_gp = gpar(fontsize = 0), column_names_gp = gpar(fontsize = 6), heatmap_legend_param = list(title = "z-score"))

```

```{r}
sessionInfo()
```


